### PR TITLE
Consolidate shard level modules without logic into IndexShardModule

### PR DESCRIPTION
**Pull request by s1monw**
We have a lot of module classes that don't contain any actual logic,
only declarative bind actions. These classes are unnecessary and can
be consolidated into the already existings IndexShardModule

### CODE REVIEW

**Comment by rjernst** [comment]
LGTM

File: src/main/java/org/elasticsearch/index/shard/IndexShardModule.java [hunk 1]
```diff
@@ -69,7 +90,25 @@ protected void configure() {
...
+        bind(ShardQueryCache.class).asEagerSingleton();
+        bind(ShardBitsetFilterCache.class).asEagerSingleton();
+        bind(ShardFieldData.class).asEagerSingleton();
+        bind(IndexShardGateway.class).asEagerSingleton();
+        bind(IndexShardGatewayService.class).asEagerSingleton();
+        bind(PercolatorQueriesRegistry.class).asEagerSingleton();
+        bind(ShardPercolateService.class).asEagerSingleton();
+        bind(ShardTermVectorsService.class).asEagerSingleton();
+        bind(IndexShardSnapshotAndRestoreService.class).asEagerSingleton();
+        bind(ShardSuggestService.class).asEagerSingleton();
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/index/shard/IndexShardModule.java]: just to clarify, this seems to be an additional binding that wasn't there before?

File: src/main/java/org/elasticsearch/index/IndexService.java [hunk 2]
```diff
@@ -36,36 +36,37 @@
...
 import org.elasticsearch.index.shard.*;
 import org.elasticsearch.index.similarity.SimilarityService;
-import org.elasticsearch.index.snapshots.IndexShardSnapshotModule;
+import org.elasticsearch.index.snapshots.IndexShardSnapshotAndRestoreService;
 import org.elasticsearch.index.store.IndexStore;
 import org.elasticsearch.index.store.Store;
 import org.elasticsearch.index.store.StoreModule;
-import org.elasticsearch.index.suggest.SuggestShardModule;
-import org.elasticsearch.index.termvectors.ShardTermVectorsModule;
+import org.elasticsearch.index.suggest.stats.ShardSuggestService;
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/index/IndexService.java]: is this import used?

File: src/main/java/org/elasticsearch/index/shard/IndexShardModule.java (hunk 1 above)
Comment (s1monw) [inline src/main/java/org/elasticsearch/index/shard/IndexShardModule.java]: ahh missed that - I just removed the class :)

**Pull request merged**